bpf/mock/helpers_skb.h

```diff
@@ -0,0 +1,106 @@
+// Header of helper functions related to skb.
+// It contains all the skb-related helper functions used in Cilium.
+// It is used to generate corresponding mock helper functions.
+// If other helper functions are needed, please add the function declarations at the bottom.
+// After adding, generate mock helper functions again with "$ruby cmock.rb -obpf.yaml helpers_skb.h".
+
+#ifndef __MOCK_HELPERS_SKB__
```

Comment:
Some of these headers seem like they're duplicated or derived from the existing versions. Is there a way to avoid duplication? How were these created? How do we keep them up to date?

Reply:
You are right, bpf/mock/helpers_skb.h and bpf/mock/helpers_xdp.h are derived from bpf/helpers.h, bpf/helpers_skb.h, and bpf/helpers_xdp.h. It could be possible to avoid these headers, but we will have to modify a lot in the Cilium source files. It will be much eaiser to list the declarations of the helper functions in these headers because they are only used to be included and compiled with mock library. And if there are any additional helper functions needed in the future, just add the declarations at the bottom.

Reply:
OK, but we should make it super clear how to update these (for instance, autogenerating this file via a make target, or having some checker that runs in CI that makes sure that this file is in sync with the one it's derived from and complains if they're not synchronized). Right now roughly ~none of the core team knows how this framework works, and I'd like to minimize the set of struggles to keep it updated & allow folks to use it in future. Otherwise it rots in the tree and when someone tries to use it, they have to solve two problems - updating the mock framework and then writing their tests. The fewer barriers to participate in this framework, the better.

Reply:
Noted. I will try to get a way to generate those headers automatically.

Reply:
Are you going to generate the full set automatically? If so, as Joe suggested, I think it makes sense to also add a target in the Makefile to generate the headers and a test in CI to check for parity between these headers and the existing kernel ones.